server/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -288,6 +288,12 @@ static void init(
         } catch (IOException e) {
             throw new BootstrapException(e);
         }
+        // a misconfigured java.io.tmpdir can cause hard-to-diagnose problems later, so reject it immediately
```

Comment:
I am good with the change, but I wonder if we should validate it sooner (e.g., in Elasticsearch). There are other components of the system that might touch the `java.io.tmpdir` before this validation is performed?

Reply:
Sure, I pushed another commit with the check moved into `Elasticsearch.execute()`.  That method seems to be the earliest one in the `Elasticsearch` class that has access to an `Environment` object.